File: Data Structures & Algorithms/validate-parentheses/submission-25.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        p = {
            ')': '(',
            ']': '[',
            '}': '{'
        }

        for c in s:
            if stack and c in p:
                if p[c] != stack[-1]:
                    return False
                stack.pop()
            else:
                stack.append(c)
        
        return len(stack) == 0

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-25.ts
class Solution {
    /**
     * @param {string[]} tokens
     * @return {number}
     */
    evalRPN(tokens: string[]): number {
        const stack: number[] = []
        const ops = new Set(['+', '-', '*', '/'])

        for (const t of tokens) {
            if (ops.has(t)) {
                const b = stack.pop()
                const a = stack.pop()
                let res = 0

                if (t === '+') {
                    res = a + b
                } else if (t === '-') {
                    res = a - b
                } else if (t === '*') {
                    res = a * b
                } else {
                    res = Math.trunc(a / b)
                }
                stack.push(res)
            } else {
                stack.push(Number(t))
            }
        }

        return stack.pop()
    }
}
